libs/dh/shared/ui-util/src/lib/dh-dropdown-translator.directive.ts

```diff
@@ -31,26 +31,33 @@ export class DhDropdownTranslatorDirective implements OnInit {
 
   translateKey = input.required<string>();
 
+  translation = signal<object | undefined>(undefined);
+
+  options = input<WattDropdownOptions>([]);
+
+  constructor() {
+    effect(() => {
+      const options = this.options();
+      const keys = this.translation();
+
+      if (!keys) return;
+
+      const translatedOptions = options.map((option) => ({
+        ...option,
+        displayValue: this.translateDisplayValue(keys[option.value as keyof typeof keys]),
+      }));
+
+      this.host.options = this.host.sortDirection
+        ? this.host.sortOptions(translatedOptions)
+        : translatedOptions;
+    });
+  }
+
   ngOnInit(): void {
     this.translocoService
```

Comment:
Suggestion: If possible i would move this to the constructor as well, and remove the ngOnInit

Reply:
I tried that, but since it is using a required input signal, I got the usual error with "signal used before it is ready" thing

Reply:
Aha, that one again !